evm/p2p/peer.py

```diff
@@ -304,16 +290,16 @@ def process_p2p_handshake(self, msg: bytes) -> None:
             raise PeerDisconnected("Peer disconnected before completing handshake: {}".format(
                 decoded_msg['reason_name']))
         remote_capabilities = decoded_msg['capabilities']
-        self.match_protocols(remote_capabilities)
-        if len(self.enabled_sub_protocols) == 0:
+        self.select_sub_protocol(remote_capabilities)
```

Comment:
The *side-effect* nature of this stood out to me.  Is there a way that we can restructure so that it does something like return the selected subprotocol and then check it locally prior to setting it to `self.sub_proto`.

```python
self.sub_proto = self.select_sub_protocol(...)
if self.sub_proto is None:
    ...
```

Reply:
Yeah, that's a nice improvement